PatreonRSS.php

```diff
@@ -69,16 +69,20 @@ public function __construct($id)
      */
     public function rss()
     {
-        $data = $this->getData();
-        echo '<?xml version="1.0"?>';
-        echo '<rss version="2.0">';
-        echo '<channel>';
-        $this->printRssChannelInfo($data['campaign'], $data['user']);
-        foreach ($data['posts'] as $item) {
-            $this->printRssItem($item);
+        if($data = $this->getData())
+        {
```

Comment:
using the guardian pattern, where you abort early, make the whole function easier to read than this huge conditional.

Reply:
It would also make more sense to throw an exception here. Otherwise empty output will be cached as RSS later resulting in invalid feeds for feed readers. An uncaught exception would at least signal a 500 error to feed readers.